modules/analytics/databricks_workspace/workspace.tf

```
@@ -48,48 +48,15 @@ resource "azurerm_databricks_workspace" "ws" {
       #   null
       # )
 
-      virtual_network_id = try(coalesce(
-        try(var.settings.custom_parameters.virtual_network_id, null),
-        try(var.vnets[var.client_config.landingzone_key][var.settings.custom_parameters.vnet_key].id, null),
-        try(var.vnets[var.settings.custom_parameters.lz_key][var.settings.custom_parameters.vnet_key].id, null),
-        ),
-        null
-      )
-
-      public_subnet_name = try(coalesce(
-        try(var.settings.custom_parameters.public_subnet_name, null),
-        try(var.vnets[var.client_config.landingzone_key][var.settings.custom_parameters.vnet_key].subnets[var.settings.custom_parameters.public_subnet_key].name, null),
-        try(var.vnets[var.settings.custom_parameters.lz_key][var.settings.custom_parameters.vnet_key].subnets[var.settings.custom_parameters.public_subnet_key].name, null),
-        ),
-        null
-      )
+      virtual_network_id = can(var.settings.custom_parameters.virtual_network_id) || can(var.settings.custom_parameters.vnet_key) == false || can(var.settings.custom_parameters.vnet.key) == false ? try(var.settings.custom_parameters.virtual_network_id, null) : var.vnets[try(var.settings.custom_parameters.lz_key, var.client_config.landingzone_key)][var.settings.custom_parameters.vnet_key].id
+      public_subnet_name = can(var.settings.custom_parameters.public_subnet_name) || can(var.settings.custom_parameters.vnet_key) == false || can(var.settings.custom_parameters.vnet.key) == false ? try(var.settings.custom_parameters.public_subnet_name, null) : var.vnets[try(var.settings.custom_parameters.lz_key, var.client_config.landingzone_key)][var.settings.custom_parameters.vnet_key].subnets[var.settings.custom_parameters.public_subnet_key].name
```

```suggestion
      public_subnet_name = can(var.settings.custom_parameters.public_subnet_name) || can(var.settings.custom_parameters.vnet_key) == false ? try(var.settings.custom_parameters.public_subnet_name, null) : var.vnets[try(var.settings.custom_parameters.lz_key, var.client_config.landingzone_key)][var.settings.custom_parameters.vnet_key].subnets[var.settings.custom_parameters.public_subnet_key].name
```

Same as previous. Removed
```
|| can(var.settings.custom_parameters.vnet.key) == false
```